src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -45,205 +45,315 @@
+        final String type = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT, null);
+        final String index = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ES_INDEX, "'security-auditlog-'YYYY.MM.dd");
+
+        return new ComplianceConfig(
+                logExternalConfig,
+                logInternalConfig,
+                logReadMetadataOnly,
+                logWriteMetadataOnly,
+                logDiffsForWrite,
+                watchedReadFields,
+                watchedWriteIndices,
+                immutableIndicesPatterns,
+                saltAsString,
+                opendistrosecurityIndex,
+                type,
+                index);
+    }
 
-	public boolean isExternalConfigLogged() {
-		return externalConfigLogged;
-	}
+    /**
+     * Checks if config defined in elasticsearch config directory must be logged
+     * @return true/false
+     */
+    public boolean shouldLogExternalConfig() {
+        return logExternalConfig;
+    }
 
-	public void setExternalConfigLogged(boolean externalConfigLogged) {
-		this.externalConfigLogged = externalConfigLogged;
-	}
+    /**
+     * Checks if internal config must be logged
+     * @return true/false
+     */
+    public boolean shouldLogInternalConfig() {
+        return logInternalConfig;
+    }
 
-	public boolean isEnabled() {
+    /**
+     * Checks if compliance is enabled
+     * @return true/false
+     */
+    public boolean isEnabled() {
         return this.enabled;
     }
 
-    //cached
-    @SuppressWarnings("unchecked")
-    private Set<String> getFieldsForIndex0(String index) {
+    /**
+     * Checks if logs diffs must be recorded for write requests
+     * Log metadata only for write requests must be disabled
+     * @return true/false
+     */
+    public boolean shouldLogDiffsForWrite() {
+        return !shouldLogWriteMetadataOnly() && logDiffsForWrite;
+    }
+
+    /**
+     * Checks if only metadata for write requests should be logged
+     * @return true/false
+     */
+    public boolean shouldLogWriteMetadataOnly() {
+        return logWriteMetadataOnly;
+    }
 
-        if(index == null) {
-            return Collections.EMPTY_SET;
+    /**
+     * Checks if only metadata for read requests should be logged
+     * @return true/false
+     */
+    public boolean shouldLogReadMetadataOnly() {
+        return logReadMetadataOnly;
+    }
+
+    /**
+     * Get set of immutable index pattern
+     * @return set of index patterns
+     */
+    public Set<String> getImmutableIndicesPatterns() {
+        return immutableIndicesPatterns;
+    }
+
+    /**
+     * Get the salt in bytes for filed anonymization
+     * @return salt in bytes
+     */
+    public byte[] getSalt16() {
+        return salt16.clone();
```

Comment:
I don't think that copy/clone is necessary, but if it is, I would rather use `Arrays.copyOf()`.

Reply:
The array can be modified by some other classes right? Requesting salt will give the modified array.

Reply:
There are no other classes that modify the array. In this particular case, if you track how salt is used, it will be copied twice.

Reply:
If you look at MaskedField where you are passing salt16, you are passing the salt to 3rd lib function like this https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/MaskedField.java#L176 

You never know if other libs modify the array or not. Is it safer to return a copy and I would prefer doing the same.

Moreover copying salt is current functionality and can changes around this if needed be done separately?